pkg/table-router/router.go

```diff
@@ -23,10 +24,59 @@ import (
 // TableRule is a rule to route schema/table to target schema/table
 // pattern format refers 'pkg/table-rule-selector'
 type TableRule struct {
-	SchemaPattern string `json:"schema-pattern" toml:"schema-pattern" yaml:"schema-pattern"`
-	TablePattern  string `json:"table-pattern" toml:"table-pattern" yaml:"table-pattern"`
-	TargetSchema  string `json:"target-schema" toml:"target-schema" yaml:"target-schema"`
-	TargetTable   string `json:"target-table" toml:"target-table" yaml:"target-table"`
+	SchemaPattern string     `json:"schema-pattern" toml:"schema-pattern" yaml:"schema-pattern"`
+	TablePattern  string     `json:"table-pattern" toml:"table-pattern" yaml:"table-pattern"`
+	TargetSchema  string     `json:"target-schema" toml:"target-schema" yaml:"target-schema"`
+	TargetTable   string     `json:"target-table" toml:"target-table" yaml:"target-table"`
+	Extractors    Extractors `json:",inline" toml:",inline" yaml:",inline"`
+}
+
+// Extractors is a rule that extracts certain values into a column
+type Extractors struct {
+	TableExtractor  *TableExtractor  `json:"extract-table" toml:"extract-table" yaml:"extract-table"`
+	SchemaExtractor *SchemaExtractor `json:"extract-schema" toml:"extract-schema" yaml:"extract-schema"`
+	SourceExtractor *SourceExtractor `json:"extract-source" toml:"extract-source" yaml:"extract-source"`
+}
+
+// TableExtractor extracts table name to column
+type TableExtractor struct {
+	TargetColumn string `json:"target-column" toml:"target-column" yaml:"target-column"`
+	TableRegexp  string `json:"table-regexp" toml:"table-regexp" yaml:"table-regexp"`
+	Regexp       *regexp.Regexp
```

Comment:
maybe `Regexp` can made private because this package has fulfilled it's functionality?